### PR TITLE
Actually run conformance tests in `just generate-conformance`.

### PR DESCRIPTION
**Pull request by rofrankel**
* Cache the Ubuntu runner (possible future improvements: install dynamically? support Windows?).
* Add a line to the `just` command that actually runs conformance tests.
* Use the release rather than debug build, because:
  * I get a stack overflow error when trying to run conformance tests with a debug build on Windows...
  * Most of the time is running the tests rather than building the tool, so the extra compilation time is likely worth it.  (But I have not actually measured this.)

### CODE REVIEW

File: Justfile [hunk 1]
```diff
@@ -4,5 +4,6 @@ default:
 generate-conformance:
 	rm -rf conformance/generated || true
 	mkdir conformance/generated
-	cargo build
-	protoc -Iconformance/protos conformance.proto test_messages_proto3.proto --luau_out=conformance/generated --plugin=protoc-gen-luau=./target/debug/protoc-gen-luau
+	cargo build --release
+	protoc -Iconformance/protos conformance.proto test_messages_proto3.proto --luau_out=conformance/generated --plugin=protoc-gen-luau=./target/release/protoc-gen-luau
+	cd conformance &&	./runner/bin/conformance_test_runner conformance.py
```
Comment (Kampfkarren) [inline Justfile:9]:
```suggestion
	cd conformance && ./runner/bin/conformance_test_runner conformance.py
```

**Comment by Kampfkarren** [comment]
Not sure what the issue on CI is--I just ran the tests locally and it only removed one failing test, didn't add any more

**Comment by rofrankel** [comment]
> Not sure what the issue on CI is--I just ran the tests locally and it only removed one failing test, didn't add any more

I think maybe it's that `cd conformance && ./runner/bin/conformance_test_runner conformance.py` is returning a nonzero exit status, which signals to GH that the step failed.  I added ` || true` to the end of the command in that step; let's see if that works.

**Comment by rofrankel** [comment]
Looks like it worked!

**Pull request merged**